### PR TITLE
ComboBox/Toggle: Gray out combobox & toggle label when component is disabled

### PR DESCRIPTION
**Pull request by montselozanod**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Before this change, when a combobox was disabled, its label had no visual indication (no change) reflecting this state. This change was happening because, even though the disabled style for a Label component already had the gray change, when creating the Label, we were not providing the disabled value to the label.

##### Before Changes

ComboBox
screen shot 2017-10-11 at 4 16 41 pm

Toggle
screen shot 2017-10-11 at 6 43 43 pm

#### After Changes

ComboBox
screen shot 2017-10-11 at 4 28 48 pm

Toggle
screen shot 2017-10-11 at 6 42 36 pm

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -246,7 +246,7 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
     return (
       <div {...divProps } ref='root' className={ this._classNames.container }>
         { label && (
-          <Label id={ id + '-label' } required={ required } htmlFor={ id } className={ this._classNames.label }>{ label }</Label>
+          <Label id={ id + '-label' } disabled={ disabled } required={ required } htmlFor={ id } className={ this._classNames.label }>{ label }</Label>
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Does this render correctly in high contrast?

Comment (montselozanod) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Let me check that 👍

**Comment by montselozanod** [comment]
Failures in screener are expected because they are failing for the disabled state, where we are now changing the text to gray.

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (montselozanod) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: screen shot 2017-10-11 at 5 04 00 pm

That is how it looks in high contrast mode... should I change the visual indication.. maye grey it out too for high contrast?

**Comment by montselozanod** [comment]
This is how high contrast looks like with the new change I pushed:

screen shot 2017-10-11 at 5 26 46 pm

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: That doesn't look good. The text should be rendering with greyText (if you are in high contrast black disabled text should be lime green)

**Comment by dzearing** [comment]
@montselozanod Nice! I noticed also Toggle doesn't disable the label. If you're in the area it might be worth fixing as well if you have a sec. (not needed for this pr)

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (montselozanod) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: I updated the PR with a fix... (see the screenshot underneath)... does that look right?

**Comment by montselozanod** [comment]
@dzearing  it's an easy fix... let me push the fix to the label of Toggle and proceed to update the PR info.

File: common/changes/office-ui-fabric-react/malozano-ComboBoxDisableStyle_2017-10-11-23-28.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "\"Make label text gray when combobox is disabled\" ",
```
Comment (dzearing) [inline common/changes/office-ui-fabric-react/malozano-ComboBoxDisableStyle_2017-10-11-23-28.json]: Minor note; when you add the change comments, make them in this format:

ComponentName: thing that changed.

Comment (montselozanod) [reply common/changes/office-ui-fabric-react/malozano-ComboBoxDisableStyle_2017-10-11-23-28.json]: Thanks for the feedback! Will do.

**Comment by dzearing** [comment]
looks good to me!

File: packages/office-ui-fabric-react/src/components/Toggle/Toggle.classNames.ts [hunk 3]
```diff
@@ -15,6 +15,8 @@ export interface IToggleClassNames {
   text?: string;
 }
 
+const MS_HIGHCONTRAST_ACTIVE = '@media screen and (-ms-high-contrast: active) &';
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Toggle/Toggle.classNames.ts]: we should put this in a common file somewhere

**Comment by phkuo** [comment]
Code looks fine but are we sure this is the right design? The text you're graying out is the description, which is helper text, which does not always get grayed out. I would say make sure design knows about the change.

**Comment by betrue-final-final** [comment]
Sorry for chiming in late, but high contrast should use system metrics and outlines. It would look like this in the default color scheme of High Contrast Black, for example:

image

That bright green is the disabled color there, and people will use that system slot to customize to their liking. Also, backgrounds tend to go away in high contrast to avoid readability issues for people with low vision.

@kristoferbrown implemented system colors for high contrast themes a bit back. Did this component not have that work done yet, Kris? @montselozanod, here's his PR: https://github.com/OfficeDev/office-ui-fabric-react/pull/2589

**Comment by kristoferbrown** [comment]
@betrue-final-final, this component should look like your image as of the PR you mentioned, but it looks like something is overwriting the border/text color to `WindowText` now.

@montselozanod, can you add `getDisabledStyles()` in the main comboBox styles so that it renders with `GrayText` borders and text in high contrast mode?

**Comment by jspurlin** [comment]
Yeah, I spoke with @montselozanod and there a few underlying issues that are causing the styling she's setting to not actually get applied. The next update should should look more expected

**Comment by betrue-final-final** [comment]
Cool, love to see the screenshots!

**Comment by montselozanod** [comment]
There were some changes that needed to be made in mergeStyles to have the correct output for media queries. Thanks @dzearing for looking at this. After said change and some tweaks in both label.classnames and toggle.classnames, this is how both components look now when they are disabled in high contrast.

combobox-highcontrast

toggle-highcontrast

Let me know what you guys think :)

**Comment by betrue-final-final** [comment]
Looks good!

**Pull request merged**